File: JAVA/Anagrams.java
public class Solution {
    public ArrayList<String> anagrams(String[] strs) {
        ArrayList<String> ans = new ArrayList<String>();
        HashMap<String, ArrayList<String>> map = new HashMap<String, ArrayList<String>>();
        for (int i = 0; i < strs.length; i++)
        {
            char[] chars = strs[i].toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);
            if (map.containsKey(key))
            {
                map.get(key).add(strs[i]);
            }
            else
            {
                ArrayList<String> list = new ArrayList<String>();
                list.add(strs[i]);
                map.put(key, list);
            }
        }
        for (String s : map.keySet())
        {
            if (map.get(s).size() > 1)
            {
                ans.addAll(map.get(s));
            }
        }
        return ans;
    }
}

File: Anagrams.cpp
class Solution {
public:
    vector<string> anagrams(vector<string> &strs) {
        unordered_map<string, vector<int>> map;
        for (int i = 0; i < strs.size(); i++) {
            string s = strs[i];
            sort(s.begin(), s.end());
            map[s].push_back(i);
        }
        vector<string> result;
        for (auto iter = map.begin(); iter != map.end(); iter++) {
            if (iter->second.size() > 1) {
                for (int i = 0; i < iter->second.size(); i++) {
                    result.push_back(strs[iter->second[i]]);
                }
            }
        }
        return result;
    }
};
